ledfx/devices/hue.py

```diff
@@ -0,0 +1,158 @@
+                description="Entertainment zone group name",
+            ): str,
+        }
+    )
+
+    status: Dict[int, Tuple[int, int, int]]
+    _sock: Optional[socket.socket] = None
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+
+        self._dtls_client_context = tls.ClientContext(
+            tls.DTLSConfiguration(
+                pre_shared_key=(
+                    self._config["user_name"],
+                    bytes.fromhex(self._config["client_key"]),
+                ),
+                ciphers=["TLS-PSK-WITH-AES-128-GCM-SHA256"],
+            )
+        )
+        self.status = {}
+
+    def _hue_get(self, api_endpoint, data=None):
+        return requests.get(
+            f"http://{self._config['ip_address']}/api/{self._config['user_name']}/{api_endpoint}",
+            json=data,
+        ).json()
+
+    def _hue_put(self, api_endpoint, data):
+        return requests.put(
+            f"http://{self._config['ip_address']}/api/{self._config['user_name']}/{api_endpoint}",
+            json=data,
+        ).json()
+
+    def _entertainment_groups(self):
+        all_groups = self._hue_get("groups")
+        return {
+            id: all_groups[id]
+            for id in all_groups
+            if all_groups[id]["type"] == "Entertainment"
+        }
+
+    def activate(self):
+        request_data = {"stream": {"active": True}}
+        response = self._hue_put(
+            f"groups/{self._config['group_id']}", request_data
+        )[0]
+        if "success" in response:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            self._sock = self._dtls_client_context.wrap_socket(sock, None)
+            self._sock.connect(
+                (self._config["ip_address"], self._config["udp_port"])
+            )
+        else:
+            raise Exception("Unable to activate UDP stream mode")
+
+        # I'm a hack, but I seem to be necessary and I work
+        for _ in range(10):
+            try:
+                time.sleep(0.2)
+                self._sock.do_handshake()
+            except exceptions.TLSError as e:
+                ...
+
+        super().activate()
+
+    def deactivate(self):
+        if self._sock is not None:
+            self._sock.close()
+            self._sock = None
+        request_data = {"stream": {"active": False}}
+        self._hue_put(f"groups/{self._config['group_id']}", request_data)
+
+        super().deactivate()
+
+    def flush(self, data):
+        pixels = [[int(r), int(g), int(b)] for r, g, b in data]
+        send_data = bytearray(b"HueStream")
+        send_data.append(1)  # Major version
+        send_data.append(0)  # Minor version
+        send_data.append(0)  # Sequence Number
+        send_data.append(0)  # Reserved
+        send_data.append(0)  # Reserved
+        send_data.append(0)  # Color Mode (0=RGB, 1=XY)
+        send_data.append(0)  # Reserved
+        for i in range(len(pixels)):
```

Comment:
Will list comprehension make this faster?

i.e

```py
send_data.extend(
    [0, *light_id.to_bytes(2, byteorder="big"), r, r, g, g, b, b]
    for i, (r, g, b) in enumerate(pixels)
    for light_id in [self._config["pixel_light_ids"][i]]
)
```

Reply:
Any suggestions on how to use this format while converting the `light_id` to an int?  The original implementation included `light_id = int(self._config["pixel_light_ids"][I])`, which isn't accounted for here.

Reply:
Yeah you know what, for now, this is perfect.